### PR TITLE
Add new ads to News landing page

### PR DESCRIPTION
**Pull request by ashleyjelks**
This PR introduces the ads served by Hashtag Labs/GAM on articles on the News Landing page.

[Links to GROW-1203](https://artsyproduct.atlassian.net/browse/GROW-1203)

Kapture 2019-05-22 at 19 11 09

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 364d5183479eaabcb9a0725a1bf45d69aab76934

https://deploy-preview-2453--artsy-reaction.netlify.com

File: src/Components/Publishing/Display/DisplayTargeting.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { data as sd } from "sharify"
+
+export const targetingData = (id: string, pageType: string) => {
+  const isTesting = sd.HASHTAG_LAB_ADS_ENABLED || "true"
+  return {
+    is_testing: JSON.parse(isTesting),
```
Comment (damassi) [inline src/Components/Publishing/Display/DisplayTargeting.tsx]: So there's no divergance between the value you're returning from the function and what's referenced internal to it, best to keep things 1:1: 

```js
const isTesting = JSON.parse(sd.HASHTAG_LAB_ADS_ENABLED || "true")
return {
  is_testing: isTesting,
  ... 
}
```

File: src/Components/Publishing/Layouts/NewsLayout.tsx [hunk 2]
```diff
@@ -74,8 +84,54 @@ export class NewsLayout extends Component<Props, State> {
     }
   }
 
+  getAdUnit() {
+    const { articleSerial, isMobile } = this.props
+
+    if (articleSerial === 3) {
```
Comment (damassi) [inline src/Components/Publishing/Layouts/NewsLayout.tsx]: Maybe a small comment above here? 3 reads like a magic number as an outside reviewer

File: src/Components/Publishing/Layouts/__tests__/NewsLayoutAdsEnabled.test.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+declare const global: any
+global.fetch = jest.fn(() =>
+  Promise.resolve({
+    status: 200,
+    json: () => Promise.resolve({}),
+  })
+)
+
+// tslint:disable: jsx-boolean-value
```
Comment (damassi) [inline src/Components/Publishing/Layouts/__tests__/NewsLayoutAdsEnabled.test.tsx]: Why is this needed?

Comment (ashleyjelks) [reply src/Components/Publishing/Layouts/__tests__/NewsLayoutAdsEnabled.test.tsx]: B/C I'm passing boolean values as props and TS complains without it

Comment (damassi) [reply src/Components/Publishing/Layouts/__tests__/NewsLayoutAdsEnabled.test.tsx]: Can you post an example? 

```
<Box foo>
```
Should work just fine (assuming `foo` is a `boolean`)

File: src/Components/Publishing/Layouts/__tests__/NewsLayoutAdsEnabled.test.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+describe("News Layout with new ads enabled", () => {
+  const getWrapper = () => {
+    return mount(
+      <NewsLayout
+        areHostedAdsEnabled={true}
+        article={NewsArticle}
+        shouldAdRender={true}
```
Comment (l2succes) [inline src/Components/Publishing/Layouts/__tests__/NewsLayoutAdsEnabled.test.tsx]: Like @damassi said you can just do `shouldAdRender` instead of `shouldAdRender={true}`

File: src/Components/Publishing/Layouts/NewsLayout.tsx [hunk 5]
```diff
@@ -74,8 +84,66 @@ export class NewsLayout extends Component<Props, State> {
     }
   }
 
+  // Ad unit code "Mobile/Desktop_InContentLB1" is for ads that appear after the 3rd article,
+  // "<Mobile/Desktop_InContentLB2" for ads that appear after the 9th article,
+  // and Mobile/Desktop_InContentLBRepeat is for all subsequent ads
```
Comment (damassi) [inline src/Components/Publishing/Layouts/NewsLayout.tsx]: #minor - conventionally multi-line comments above class methods (and functions) are in the dockblock format:

```
/**
 * Ad unit code "Mobile/Desktop_InContentLB1" is for ads that appear after the 
 * 3rd article, "<Mobile/Desktop_InContentLB2" for ads that appear after the 9th 
 * article, and Mobile/Desktop_InContentLBRepeat is for all subsequent ads
 */
```
In VSCode, if you type `/**` and hit enter it should automatically expand. 

That said, it would be better to put the comment above the if branch where the 3 and the 9 live, rather than at the top. People reading the logic inside of the function will appreciate it later.

**Comment by artsyit** [comment]
:rocket: PR was released in v16.12.13 :rocket:
